w3af/core/data/parsers/doc/wsdl.py

```diff
@@ -19,194 +19,110 @@
 Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 
 """
+import contextlib
+import sys
 import xml.parsers.expat as expat
+from cStringIO import StringIO
 
 import SOAPpy
+import zeep
+from requests import HTTPError
+from zeep.exceptions import XMLSyntaxError
 
 import w3af.core.controllers.output_manager as om
+import w3af.core.data.kb.knowledge_base as kb
 from w3af.core.controllers.exceptions import BaseFrameworkException
+from w3af.core.data.kb.info import Info
+from w3af.core.data.parsers.doc.baseparser import BaseParser
 from w3af.core.data.parsers.doc.url import URL
+from w3af.core.controllers import output_manager
 
 
-class WSDLParser(object):
+class WSDLParser(BaseParser):
     """
     This class parses WSDL documents.
 
     :author: Andres Riancho (andres.riancho@gmail.com)
     """
 
-    def __init__(self):
+    def __init__(self, http_response):
         self._proxy = None
-
-    def is_WSDL(self, data):
-        """
-        This is not a 100% accurate test, the real WSDL parsing is performed
-        in "SOAPpy.WSDL.Proxy( xmlData )". This test was mostly added to
-        enhance framework's performance.
-
-        :param data: A string that might represent a WSDL
-        :return: True if the data parameter is a WSDL document.
-        """
-        return False
-        if '<definitions' in data[:150] or '<wsdl:definitions' in data[:150]:
-            return True
-        else:
+        super(WSDLParser, self).__init__(http_response)
+        self._wsdl_client = zeep.Client(str(http_response.get_uri()))
+        self._discovered_urls = set()
+
+    def __getstate__(self):
+        state = self.__dict__.copy()
+        del state['_wsdl_client']
+        return state
+
+    def __setstate__(self, state):
+        self.__dict__.update(state)
+        self._wsdl_client = zeep.Client(str(self._http_response.get_uri()))
+
+    @staticmethod
+    def can_parse(http_resp):
+        url = http_resp.get_uri()
+        try:
+            wsdl_client = zeep.Client(str(url))
```

Comment:
This will send an HTTP request to the URL *without* using the w3af HTTP client. This means that, for example, HTTP proxy configurations are not going to be respected. For some cases that will break the feature. Is there a way to create a wsdl_client by sending the WSDL URL and HTTP response content?

Reply:
fix in https://github.com/andresriancho/w3af/pull/18645/commits/9ba5a080b1cb755215d68dd1b197c1a705b2c591